internal/git/first.go

```diff
@@ -0,0 +1,28 @@
+package git
+
+import (
+	"fmt"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func FirstCommit(repoPath string) (string, error) {
+	r, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return "", fmt.Errorf("unable to open repo: %w", err)
+	}
+	iter, err := r.Log(&git.LogOptions{})
+	if err != nil {
+		return "", fmt.Errorf("unable to log commits: %w", err)
+	}
+
+	var last string
+	err = iter.ForEach(func(c *object.Commit) error {
+		if c != nil {
+			last = c.Hash.String()
+		}
+		return nil
+	})
+	return last, err
```

Comment:
nit: the term `last` is confusing in the context of a _first_ commit function. I understand this is correct because the commits are listed in a reverse chronological order.

Reply:
This confused me too; maybe a rename or a comment.

Reply:
> I understand this is correct because the commits are listed in a reverse chronological order.

exactly, has the same semantics as git log. I'll add a comment